core/src/main/scala/org/apache/spark/mapred/SparkHadoopMapRedUtil.scala

```diff
@@ -91,8 +91,7 @@ object SparkHadoopMapRedUtil extends Logging {
       committer: MapReduceOutputCommitter,
       mrTaskContext: MapReduceTaskAttemptContext,
       jobId: Int,
-      splitId: Int,
-      attemptId: Int): Unit = {
```

Comment:
Here, I removed this parameter because we should always get the attempt number from the TaskContext in order to ensure that it's correct. This had a ripple-effect on a few other callsites, which I've updated.